apps/angular/src/app/features/anime/anime-management/anime-form/anime-form.component.ts

```diff
@@ -0,0 +1,165 @@
+
+	/** Type. */
+	readonly type: FormControl<AnimeType>;
+
+	/** Status. */
+	readonly status: FormControl<AnimeStatus>;
+
+	/** Provides anime is airing. */
+	readonly isAiring: FormControl<boolean>;
+
+	/** Description. */
+	readonly description: FormControl<string>;
+
+	/** Age rating. */
+	readonly ageRating: FormControl<AnimeRating>;
+
+	/** Source. */
+	readonly source: FormControl<AnimeSource>;
+
+	/** Season. */
+	readonly season: FormControl<AnimeSeason>;
+
+	/** Aired start date. */
+	readonly airedStart: FormControl<Date>;
+
+	/** Aired end date. */
+	readonly airedEnd: FormControl<Date>;
+
+	/** Image file. */
+	readonly imageFile: FormControl<ImageFile>;
+
+	/** Youtube trailer id. */
+	readonly youtubeTrailerId: FormControl<string | null>;
+
+	/** Genres. */
+	readonly genres: FormControl<MultipleAutocompleteItem[]>;
+
+	/** Genres. */
+	readonly studios: FormControl<MultipleAutocompleteItem[]>;
+}
+
+/** Anime form component. */
+@Component({
+	selector: 'camp-anime-form',
+	templateUrl: './anime-form.component.html',
+	styleUrls: ['./anime-form.component.css'],
+	changeDetection: ChangeDetectionStrategy.OnPush,
+	providers: [
+		{
+			provide: MultipleAutocompleteService,
+			useClass: AnimeMultipleAutocompleteService,
+		},
+	],
+})
+export class AnimeFormComponent implements OnInit {
+
+	private readonly formBuilder = inject(NonNullableFormBuilder);
+
+	/** Anime types. */
+	protected readonly animeTypes = EnumUtils.toArray(AnimeType);
+
+	/** Anime statuses. */
+	protected readonly animeStatuses = EnumUtils.toArray(AnimeStatus);
+
+	/** Anime ratings. */
+	protected readonly animeRatings = EnumUtils.toArray(AnimeRating);
+
+	/** Anime sources. */
+	protected readonly animeSources = EnumUtils.toArray(AnimeSource);
+
+	/** Anime seasons. */
+	protected readonly animeSeasons = EnumUtils.toArray(AnimeSeason);
+
+	/** Anime form data group. */
+	protected readonly formGroup: FormGroup<AnimeManagementFormControls>;
+
+	/** Multiple autocomplete groups. */
+	protected get multipleAutocompleteGroups(): typeof animeMultipleAutocompleteGroups {
+		return animeMultipleAutocompleteGroups;
+	}
+
+	/** Anime form data. */
+	@Input()
+	public animeFormData: AnimeFormData | null = null;
+
+	/** Submit event. */
+	@Output()
+	public submitEvent = new EventEmitter<AnimeFormData>();
+
+	public constructor() {
+		this.formGroup = this.initAnimeForm();
+	}
+
+	/** @inheritdoc */
+	public ngOnInit(): void {
+		if (this.animeFormData !== null) {
+			this.formGroup.patchValue(this.animeFormData);
+		}
+	}
+
+	/**
+	 * Track by index.
+	 * @param index Item index.
+	 */
+	protected trackByIndex(index: number): number {
+		return index;
+	}
+
+	/** Handle anime management form submitting. */
+	protected handleSubmit(): void {
+		this.formGroup.markAllAsTouched();
+
+		if (this.formGroup.invalid) {
+			return;
+		}
+
+		const formData = this.formGroup.getRawValue();
+
+		this.submitEvent.emit({
+			...formData,
+		});
```

Comment:
Object destruction seems redundant here, we can remove it:
```suggestion
		this.submitEvent.emit(formData);
```